.ebextensions/99datadog.config

```diff
@@ -0,0 +1,34 @@
+# .ebextensions/99datadog.config
+option_settings:
+    - namespace: aws:elasticbeanstalk:application:environment
+      option_name: DD_AGENT_MAJOR_VERSION
+      value: "7"
+    - namespace: aws:elasticbeanstalk:application:environment
+      option_name: DD_AGENT_MINOR_VERSION
+      value: "" # Eg: 34 to install 7.34.x, leave empty to install the latest 7.x
+
+files:
+  "/etc/datadog-agent/datadog.yaml":
+    mode: "000777"
+    owner: root # will be changed to dd-agent after the installation
+    group: root
+    content: |
+      # Add here the Agent configuration
+      site: datadoghq.com
+      api_key: DD_API_KEY
+      logs_enabled: "true"
+      apm_config:
+        enabled: "true"
+        apm_non_local_traffic: "true"
+
+  "/datadog_install_script.sh":
+    mode: "000700"
+    owner: root
+    group: root
+    source: https://s3.amazonaws.com/dd-agent/scripts/install_script.sh
+
+container_commands:
+    04patch_api_key:
+        command: "sed -i.bak \"s/DD_API_KEY$/$DD_API_KEY/\" /etc/datadog-agent/datadog.yaml"
```

Comment:
Datadog recommend using their [secrets management system](https://docs.datadoghq.com/agent/guide/secrets-management/?tab=linux#helper-scripts-for-autodiscovery) for passing in the api key and other config options. Seeing as it's only the api key that we need I think this is simpler, it does require setting the api key in the Elastic Beanstalk configuration which is not ideal, but we already keep the transcrypt secret there which is the key to all other secrets so it's no less secure than the current setup